assets/js/alerts.js

```diff
@@ -0,0 +1,228 @@
+const applicationFilter = document.getElementById('application')
+const environmentFilter = document.getElementById('environment')
+const namespaceFilter = document.getElementById('namespace')
+const severityFilter = document.getElementById('severity')
+let previousDataJSON = ''
+
+jQuery(async function () {
+  //check url to see if any filters are currently applied
+  let currentFilters = getFiltersFromURL()
+  //get alerts from the api
+  const alerts = await getAlerts()
+  //use filters and alerts to update alert table and filters
+  updateAll(alerts, currentFilters)
+
+  //on click of any 'Update' button to apply filters
+  $('#updateApplicationName,#updateEnvironment,#updateNamespace,#updateSeverityLabel').on('click', async e => {
+    e.preventDefault(e)
+
+    //check which filter has been updated
+    let dropDownType = ''
+    switch (e.target.id) {
+      case 'updateApplicationName':
+        dropDownType = 'application'
+        break
+      case 'updateEnvironment':
+        dropDownType = 'environment'
+        break
+      case 'updateNamespace':
+        dropDownType = 'namespace'
+        break
+      case 'updateSeverityLabel':
+        dropDownType = 'severity'
+        break
+      default:
+        return false
+    }
+    //get the selectec dropdown option
+    const dropDownText = $(`#${dropDownType} option:selected`).text()
+
+    //update current filters
+    currentFilters[`${dropDownType}`] = dropDownText
+    //update alert table and filters
+    updateAll(alerts, currentFilters)
+
+    // Watch function updates Alerts on a timeout
+    watch()
+  })
+
+  //on click of 'Reset Filters' button, clear filters, reset url params and update table and filters
+  $('#resetFilters').on('click', async e => {
+    const params = new URLSearchParams()
+    history.replaceState(null, '', '/alerts')
+    currentFilters = {
+      application: '',
+      environment: '',
+      namespace: '',
+      severity: '',
+    }
+    updateDropdowns(alerts, currentFilters)
+    updateAll(alerts, currentFilters)
+  })
+})
+
+//gets alerts every 5 seconds
+const watch = async () => {
+  await updateAlerts()
+
+  setTimeout(watch, 5000)
+}
+
+const updateAlerts = async () => {
+  try {
+    const currentData = await getAlerts()
+    //check against existing alerts
+    if (currentData !== previousDataJSON) {
+      alerts = currentData
```

Comment:
These should be passed in as args as we don't want the function changing data outside it's scope, same for `previousDataJSON`